src/utilities/poisson.jl

```diff
@@ -0,0 +1,14 @@
+"""
+    poissonST(l::AbstractVector)
+
+Generates Poisson Spike Trains based on the normalized vector. Each
+pseudo-neuron (probability p in vector), fires with probability p at each
+timestep of simulation.
+
+# Inputs
+- `l`: array of values
+"""
+
+function poissonST(l::AbstractVector)
+    return poissonST(t) = [Float64(rand(Bernoulli(p), 1)[1]) for p in normalize(l)]
```

Comment:
I have a few ideas about this implementation. 

First, I don't think returning a function with the same name is the best practice. Maybe the outer function should be called `prob2neurons` for mapping probabilities to a population of neurons which fire with some probability, and the inner function could be called `getpoissonST`. When debugging, it's a bit easier if the names don't overlap, but functionally it will work the same.

Next, the **returned** function does a lot of work every time you call it. It normalizes the probabilities, it generates the random variables, it allocates the array, and it also samples those random variables. A lot of that could be done ahead of time, like pre-allocating the array, normalizing the vector of probabilities, and generating all your random variables. 

As the length of `l` grows, allocations start to cost you a lot, so it'd be good to do them ahead of time. Ultimately, what I think you want to return here would be a "closure" over an array that you pre-allocate and the Bernoulli random variables you generate. If you read up on closures in the Julia language, that's the type of thing I think would go well here. They're mentioned in the Julia manual [here](https://docs.julialang.org/en/v1/devdocs/functions/#Closures), but it's a pretty short explanation.

Reply:
I'm not sure leaving this as a function of time is useful, because calling the function with the same time step multiple times isn't guaranteed to return the same value. Instead, maybe it's best to leave it as a function with no arguments, and it just generates the random string for you. As above, this works functionally the same, but there's no confusion about the role that `t` plays since it doesn't affect the calculation

Reply:
Last, does the returned vector need to be `Float64`? Julia might handle the arithmetic with `Bool`s just fine, since it's essential multiplying by `0` or `1`. If it doesn't work leaving it with a `Bool`, if everything will always be either a `0` or a `1`, `Int` arithmetic should be faster than `Float64` arithmetic. This is probably the least important issue to address, since in practice everything may get promoted to a `Float64` anyway. But, if a vector of `Bool`s works, that's probably the fastest solution.